app/scripts/lib/multichain-api/provider-request.js

```diff
@@ -13,16 +13,17 @@ export async function providerRequestHandler(
     return end(new Error('missing CAIP-25 endowment'));
   }
 
+  const chainId = scope.split(':')[1];
```

Comment:
we should try to use our CAIP type helpers

https://github.com/MetaMask/utils/blob/main/src/caip-types.ts#L114